### PR TITLE
id token

### PR DESCRIPTION
**Pull request by sambae**
This PR updates the library to request id token and extract it from the form body.

### CODE REVIEW

**Review by tmtrademarked — APPROVED** [review]

File: signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt [hunk 1]
```diff
@@ -71,7 +70,7 @@ class SignInWithAppleService(
                 val authenticationUri = Uri
                     .parse("https://appleid.apple.com/auth/authorize")
                     .buildUpon().apply {
-                        appendQueryParameter("response_type", "code")
+                        appendQueryParameter("response_type", "code id_token")
```
Comment (yseredba) [inline signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt:73]: Not sure what's going on here.
Should be `code id_token` or `code_id_token` or `code ` + `idTokenValue`?

File: sample/src/main/java/com/willowtreeapps/signinwithapple/sample/SampleJavaActivity.java [hunk 2]
```diff
@@ -35,8 +35,8 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         SignInWithAppleCallback callback = new SignInWithAppleCallback() {
             @Override
-            public void onSignInWithAppleSuccess(@NonNull String authorizationCode) {
-                Toast.makeText(SampleJavaActivity.this, authorizationCode, LENGTH_SHORT).show();
+            public void onSignInWithAppleSuccess(@NonNull String authorizationCode, @NonNull String idToken) {
+                Toast.makeText(SampleJavaActivity.this, authorizationCode + " " + idToken, LENGTH_SHORT).show();
```
Comment (yseredba) [inline sample/src/main/java/com/willowtreeapps/signinwithapple/sample/SampleJavaActivity.java:39]: Why we show auth code and token in toast?

**Review by yseredba — APPROVED** [review]
I cannot review it properly to be fair since I don't know details of this sign in logic.
Left some comments/questions.

File: signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt (hunk 1 above)
Comment (sambae) [reply signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt:73]: `code id_token` is correct!

File: sample/src/main/java/com/willowtreeapps/signinwithapple/sample/SampleJavaActivity.java (hunk 2 above)
Comment (sambae) [reply sample/src/main/java/com/willowtreeapps/signinwithapple/sample/SampleJavaActivity.java:39]: This is just a sample activity for the library

**Pull request merged**